### PR TITLE
Include `countme` crate to count important data structures.

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

**Comment by matklad** [comment]
```
workspaces:
186 packages loaded

analysis:
3579 (37mb) files
0 (0b) index symbols
2537 trees, 128 retained
30857 trees, 128 retained (macros)
186kb total

counts:
cnt:   total   max_live       live
hir_def::item_tree::ItemTree:
       29165      27989      27988
hir_def::nameres::DefMap:
        1413        681        678
```

File: crates/hir_def/src/item_tree.rs [hunk 1]
```diff
@@ -116,7 +119,12 @@ impl ItemTree {
     }
 
     fn empty() -> Self {
-        Self { top_level: Default::default(), attrs: Default::default(), data: Default::default() }
+        Self {
+            _c: Count::new(),
```
Comment (kjeremy) [inline crates/hir_def/src/item_tree.rs:123]: Can we `Default::default` this too?

File: crates/hir_def/src/nameres.rs [hunk 2]
```diff
@@ -75,7 +76,11 @@ use crate::{
 /// Contains all top-level defs from a macro-expanded crate
 #[derive(Debug, PartialEq, Eq)]
 pub struct DefMap {
+<<<<<<< HEAD
```
Comment (lnicola) [inline crates/hir_def/src/nameres.rs:79]: Conflict marker

File: crates/hir_def/src/nameres.rs [hunk 3]
```diff
@@ -75,7 +76,11 @@ use crate::{
...
     parent: Option<Arc<DefMap>>,
+=======
+    _c: Count<Self>,
+>>>>>>> 783c78576 (Include `countme` crate to count important data structures.)
```
Comment (kjeremy) [inline crates/hir_def/src/nameres.rs:83]: Looks like some conflicts got checked in.

File: crates/hir_def/src/item_tree.rs (hunk 1 above)
Comment (matklad) [reply crates/hir_def/src/item_tree.rs:123]: Actually, I wonder if we should just `derive(Default)`, as per style guide?

**Comment by lnicola** [comment]
```
cnt:   total   max_live       live
hir_def::item_tree::ItemTree:
       44572      44572      44572
hir_def::nameres::DefMap:
         679        679        679
rowan::cursor::SyntaxNode:
    73077405      13183          0
rowan::green::node::GreenNode:
     9525182     400664     344005
rowan::green::token::GreenToken:
     4781967     375875     374458
```

Did we grow that much since https://github.com/rust-analyzer/rust-analyzer/pull/7378#issuecomment-764767627?

**Comment by matklad** [comment]
could that be item trees for blocks?

**Comment by lnicola** [comment]
Maybe? I meant specifically `SyntaxNode` which increased from `56 622 630` to `73 077 405` and `GreenToken`, from `7 313 818` to `9 525 182`.

**Comment by matklad** [comment]
@lnicola yeah, we'd need to track that... Could you bisect?

**Comment by lnicola** [comment]
I get the new numbers even on `e5c5c0a040e0c74892ea0a36c7fd50e5410879bd`, which introduced `countme`. So maybe it was #7375.

**Comment by lnicola** [comment]
Or not -- it's the same even on #7372 with this one cherry-picked.

**Pull request merged**